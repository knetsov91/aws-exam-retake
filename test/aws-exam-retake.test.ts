// import * as cdk from 'aws-cdk-lib';
// import { Template } from 'aws-cdk-lib/assertions';
// import * as AwsExamRetake from '../lib/aws-exam-retake-stack';

// example test. To run these tests, uncomment this file along with the
// example resource in lib/aws-exam-retake-stack.ts
test('SQS Queue Created', () => {
//   const app = new cdk.App();
//     // WHEN
//   const stack = new AwsExamRetake.AwsExamRetakeStack(app, 'MyTestStack');
//     // THEN
//   const template = Template.fromStack(stack);

//   template.hasResourceProperties('AWS::SQS::Queue', {
//     VisibilityTimeout: 300
//   });
});
