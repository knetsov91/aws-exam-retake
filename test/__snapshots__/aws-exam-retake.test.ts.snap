// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test CDK Stack 1`] = `
{
  "Outputs": {
    "InventoryManagementApiEndpoint00E39A33": {
      "Value": {
        "Fn::Join": [
          "",
          [
            "https://",
            {
              "Ref": "InventoryManagementApi91CBC5B5",
            },
            ".execute-api.",
            {
              "Ref": "AWS::Region",
            },
            ".",
            {
              "Ref": "AWS::URLSuffix",
            },
            "/",
            {
              "Ref": "InventoryManagementApiDeploymentStageprodD7A24242",
            },
            "/",
          ],
        ],
      },
    },
  },
  "Resources": {
    "InventoryManagementApi91CBC5B5": {
      "Properties": {
        "Name": "InventoryManagementApi",
      },
      "Type": "AWS::ApiGateway::RestApi",
    },
    "InventoryManagementApiAccount483D05E9": {
      "DeletionPolicy": "Retain",
      "DependsOn": [
        "InventoryManagementApi91CBC5B5",
      ],
      "Properties": {
        "CloudWatchRoleArn": {
          "Fn::GetAtt": [
            "InventoryManagementApiCloudWatchRoleF10F59B2",
            "Arn",
          ],
        },
      },
      "Type": "AWS::ApiGateway::Account",
      "UpdateReplacePolicy": "Retain",
    },
    "InventoryManagementApiCloudWatchRoleF10F59B2": {
      "DeletionPolicy": "Retain",
      "Properties": {
        "AssumeRolePolicyDocument": {
          "Statement": [
            {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": {
                "Service": "apigateway.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": [
          {
            "Fn::Join": [
              "",
              [
                "arn:",
                {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AmazonAPIGatewayPushToCloudWatchLogs",
              ],
            ],
          },
        ],
      },
      "Type": "AWS::IAM::Role",
      "UpdateReplacePolicy": "Retain",
    },
    "InventoryManagementApiDeployment709CC56Df8e14c89f41564e8f2f03f4af8b25ab8": {
      "DependsOn": [
        "InventoryManagementApigetProductPOSTA91A3181",
        "InventoryManagementApigetProduct7C2824EA",
      ],
      "Properties": {
        "Description": "Automatically created by the RestApi construct",
        "RestApiId": {
          "Ref": "InventoryManagementApi91CBC5B5",
        },
      },
      "Type": "AWS::ApiGateway::Deployment",
    },
    "InventoryManagementApiDeploymentStageprodD7A24242": {
      "DependsOn": [
        "InventoryManagementApiAccount483D05E9",
      ],
      "Properties": {
        "DeploymentId": {
          "Ref": "InventoryManagementApiDeployment709CC56Df8e14c89f41564e8f2f03f4af8b25ab8",
        },
        "RestApiId": {
          "Ref": "InventoryManagementApi91CBC5B5",
        },
        "StageName": "prod",
      },
      "Type": "AWS::ApiGateway::Stage",
    },
    "InventoryManagementApigetProduct7C2824EA": {
      "Properties": {
        "ParentId": {
          "Fn::GetAtt": [
            "InventoryManagementApi91CBC5B5",
            "RootResourceId",
          ],
        },
        "PathPart": "getProduct",
        "RestApiId": {
          "Ref": "InventoryManagementApi91CBC5B5",
        },
      },
      "Type": "AWS::ApiGateway::Resource",
    },
    "InventoryManagementApigetProductPOSTA91A3181": {
      "Properties": {
        "AuthorizationType": "NONE",
        "HttpMethod": "POST",
        "Integration": {
          "IntegrationHttpMethod": "POST",
          "Type": "AWS_PROXY",
          "Uri": {
            "Fn::Join": [
              "",
              [
                "arn:",
                {
                  "Ref": "AWS::Partition",
                },
                ":apigateway:",
                {
                  "Ref": "AWS::Region",
                },
                ":lambda:path/2015-03-31/functions/",
                {
                  "Fn::GetAtt": [
                    "QueryFunctionBDF4DE5B",
                    "Arn",
                  ],
                },
                "/invocations",
              ],
            ],
          },
        },
        "ResourceId": {
          "Ref": "InventoryManagementApigetProduct7C2824EA",
        },
        "RestApiId": {
          "Ref": "InventoryManagementApi91CBC5B5",
        },
      },
      "Type": "AWS::ApiGateway::Method",
    },
    "InventoryManagementApigetProductPOSTApiPermissionMyTestStackInventoryManagementApi04CE0871POSTgetProductC5AFDE41": {
      "Properties": {
        "Action": "lambda:InvokeFunction",
        "FunctionName": {
          "Fn::GetAtt": [
            "QueryFunctionBDF4DE5B",
            "Arn",
          ],
        },
        "Principal": "apigateway.amazonaws.com",
        "SourceArn": {
          "Fn::Join": [
            "",
            [
              "arn:",
              {
                "Ref": "AWS::Partition",
              },
              ":execute-api:",
              {
                "Ref": "AWS::Region",
              },
              ":",
              {
                "Ref": "AWS::AccountId",
              },
              ":",
              {
                "Ref": "InventoryManagementApi91CBC5B5",
              },
              "/",
              {
                "Ref": "InventoryManagementApiDeploymentStageprodD7A24242",
              },
              "/POST/getProduct",
            ],
          ],
        },
      },
      "Type": "AWS::Lambda::Permission",
    },
    "InventoryManagementApigetProductPOSTApiPermissionTestMyTestStackInventoryManagementApi04CE0871POSTgetProduct707F3B74": {
      "Properties": {
        "Action": "lambda:InvokeFunction",
        "FunctionName": {
          "Fn::GetAtt": [
            "QueryFunctionBDF4DE5B",
            "Arn",
          ],
        },
        "Principal": "apigateway.amazonaws.com",
        "SourceArn": {
          "Fn::Join": [
            "",
            [
              "arn:",
              {
                "Ref": "AWS::Partition",
              },
              ":execute-api:",
              {
                "Ref": "AWS::Region",
              },
              ":",
              {
                "Ref": "AWS::AccountId",
              },
              ":",
              {
                "Ref": "InventoryManagementApi91CBC5B5",
              },
              "/test-invoke-stage/POST/getProduct",
            ],
          ],
        },
      },
      "Type": "AWS::Lambda::Permission",
    },
    "InventoryTableFD135387": {
      "DeletionPolicy": "Retain",
      "Properties": {
        "AttributeDefinitions": [
          {
            "AttributeName": "productId",
            "AttributeType": "S",
          },
        ],
        "BillingMode": "PAY_PER_REQUEST",
        "KeySchema": [
          {
            "AttributeName": "productId",
            "KeyType": "HASH",
          },
        ],
        "StreamSpecification": {
          "StreamViewType": "NEW_AND_OLD_IMAGES",
        },
      },
      "Type": "AWS::DynamoDB::Table",
      "UpdateReplacePolicy": "Retain",
    },
    "ProcessFunction7E4ECD78": {
      "DependsOn": [
        "ProcessFunctionServiceRoleDefaultPolicy57489565",
        "ProcessFunctionServiceRole101ED8D0",
      ],
      "Properties": {
        "Code": {
          "S3Bucket": {
            "Fn::Sub": "cdk-hnb659fds-assets-\${AWS::AccountId}-\${AWS::Region}",
          },
          "S3Key": "ec556379bc8aac0699f7b26924d302f6e380957cf6085bfb725ca421270a6528.zip",
        },
        "Environment": {
          "Variables": {
            "TABLE_NAME": {
              "Ref": "InventoryTableFD135387",
            },
          },
        },
        "Handler": "index.handler",
        "Role": {
          "Fn::GetAtt": [
            "ProcessFunctionServiceRole101ED8D0",
            "Arn",
          ],
        },
        "Runtime": "nodejs20.x",
      },
      "Type": "AWS::Lambda::Function",
    },
    "ProcessFunctionServiceRole101ED8D0": {
      "Properties": {
        "AssumeRolePolicyDocument": {
          "Statement": [
            {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": [
          {
            "Fn::Join": [
              "",
              [
                "arn:",
                {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
              ],
            ],
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "ProcessFunctionServiceRoleDefaultPolicy57489565": {
      "Properties": {
        "PolicyDocument": {
          "Statement": [
            {
              "Action": [
                "dynamodb:BatchGetItem",
                "dynamodb:GetRecords",
                "dynamodb:GetShardIterator",
                "dynamodb:Query",
                "dynamodb:GetItem",
                "dynamodb:Scan",
                "dynamodb:ConditionCheckItem",
                "dynamodb:BatchWriteItem",
                "dynamodb:PutItem",
                "dynamodb:UpdateItem",
                "dynamodb:DeleteItem",
                "dynamodb:DescribeTable",
              ],
              "Effect": "Allow",
              "Resource": [
                {
                  "Fn::GetAtt": [
                    "InventoryTableFD135387",
                    "Arn",
                  ],
                },
                {
                  "Ref": "AWS::NoValue",
                },
              ],
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "ProcessFunctionServiceRoleDefaultPolicy57489565",
        "Roles": [
          {
            "Ref": "ProcessFunctionServiceRole101ED8D0",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "QueryFunctionBDF4DE5B": {
      "DependsOn": [
        "QueryFunctionServiceRoleDefaultPolicyF63D0CC9",
        "QueryFunctionServiceRoleF5EB059D",
      ],
      "Properties": {
        "Code": {
          "S3Bucket": {
            "Fn::Sub": "cdk-hnb659fds-assets-\${AWS::AccountId}-\${AWS::Region}",
          },
          "S3Key": "981c820f3d41df96a0a58ce772041c25542ab61d313db63c9ca02726c2d962b6.zip",
        },
        "Environment": {
          "Variables": {
            "TABLE_NAME": {
              "Ref": "InventoryTableFD135387",
            },
          },
        },
        "Handler": "index.handler",
        "Role": {
          "Fn::GetAtt": [
            "QueryFunctionServiceRoleF5EB059D",
            "Arn",
          ],
        },
        "Runtime": "nodejs20.x",
      },
      "Type": "AWS::Lambda::Function",
    },
    "QueryFunctionServiceRoleDefaultPolicyF63D0CC9": {
      "Properties": {
        "PolicyDocument": {
          "Statement": [
            {
              "Action": [
                "dynamodb:BatchGetItem",
                "dynamodb:GetRecords",
                "dynamodb:GetShardIterator",
                "dynamodb:Query",
                "dynamodb:GetItem",
                "dynamodb:Scan",
                "dynamodb:ConditionCheckItem",
                "dynamodb:DescribeTable",
              ],
              "Effect": "Allow",
              "Resource": [
                {
                  "Fn::GetAtt": [
                    "InventoryTableFD135387",
                    "Arn",
                  ],
                },
                {
                  "Ref": "AWS::NoValue",
                },
              ],
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "QueryFunctionServiceRoleDefaultPolicyF63D0CC9",
        "Roles": [
          {
            "Ref": "QueryFunctionServiceRoleF5EB059D",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "QueryFunctionServiceRoleF5EB059D": {
      "Properties": {
        "AssumeRolePolicyDocument": {
          "Statement": [
            {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": [
          {
            "Fn::Join": [
              "",
              [
                "arn:",
                {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
              ],
            ],
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "ThresholdTopic006DBD27": {
      "Properties": {
        "TopicName": "thresholdTopic",
      },
      "Type": "AWS::SNS::Topic",
    },
    "ThresholdTopicSubscriptionB8346000": {
      "Properties": {
        "Endpoint": "knetsov91@gmail.com",
        "Protocol": "email",
        "TopicArn": {
          "Ref": "ThresholdTopic006DBD27",
        },
      },
      "Type": "AWS::SNS::Subscription",
    },
    "myschedule": {
      "Properties": {
        "FlexibleTimeWindow": {
          "Mode": "OFF",
        },
        "GroupName": "SchedulesForLambda",
        "ScheduleExpression": "rate(1 minute)",
        "Target": {
          "Arn": {
            "Fn::GetAtt": [
              "ProcessFunction7E4ECD78",
              "Arn",
            ],
          },
          "RoleArn": {
            "Fn::GetAtt": [
              "schedulerrole9B80A9F3",
              "Arn",
            ],
          },
        },
      },
      "Type": "AWS::Scheduler::Schedule",
    },
    "schedulegroup": {
      "Properties": {
        "Name": "SchedulesForLambda",
      },
      "Type": "AWS::Scheduler::ScheduleGroup",
    },
    "schedulepolicyB4BD75DE": {
      "Properties": {
        "PolicyDocument": {
          "Statement": [
            {
              "Action": "lambda:InvokeFunction",
              "Effect": "Allow",
              "Resource": {
                "Fn::GetAtt": [
                  "ProcessFunction7E4ECD78",
                  "Arn",
                ],
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "ScheduleToInvokeLambdas",
        "Roles": [
          {
            "Ref": "schedulerrole9B80A9F3",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "schedulerrole9B80A9F3": {
      "Properties": {
        "AssumeRolePolicyDocument": {
          "Statement": [
            {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": {
                "Service": "scheduler.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
      },
      "Type": "AWS::IAM::Role",
    },
  },
}
`;
